reconcile/utils/promotion_state.py

```diff
@@ -23,29 +23,29 @@ class Config:
         extra = Extra.forbid
 
 
-class DeploymentState:
+class PromotionState:
     """
     A wrapper around a reconcile.utils.state.State object.
-    This is dedicated to retrieving information about
-    deployments from S3.
+    This is dedicated to storing and retrieving information
+    about promotions on S3.
     """
 
     def __init__(self, state: State):
         self._state = state
         self._commits_by_channel: dict[str, set[str]] = defaultdict(set)
-        self._fetch_deployment_list()
+        self._fetch_promotions_list()
 
-    def _fetch_deployment_list(self) -> None:
+    def _fetch_promotions_list(self) -> None:
         all_keys = self._state.ls()
         for commit in all_keys:
             if not commit.startswith("/promotions/"):
                 continue
             parts = commit.split("/")
             self._commits_by_channel[parts[2]].add(parts[3])
```

Comment:
I know this is existing code, and not part of this PR. Nevertheless, could you add a comment, or make the pars named variables?

Reply:
Good point! Names the parts now and added a comment about the expected format